async-embedded/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+//! Proof of Concept async runtime for the Cortex-M architecture
+
+#![deny(missing_docs)]
+#![deny(rust_2018_idioms)]
+#![deny(warnings)]
+#![no_std]
+
+mod alloc;
+mod executor;
+pub mod task;
+pub mod unsync;
+
+#[cfg(feature = "isa-cortex-m")]
```

Comment:
I'd rather replace this with architecture autodetection (with built-in cfgs or build.rs)

Reply:
Unfortunately, we can't do that on stable yet. I need to conditionally depend on `cortex-m`, `ccortex-m-udf` and `riscv` crates. The fix for https://github.com/rust-lang/cargo/issues/2524 has landed in nightly-2020-02-23.

Reply:
I mean, I can change the `#[cfg(...)]` stuff but the user would still have to specify `features = ["isa-..."]` when depending on the crate.

Reply:
Once https://github.com/rust-lang/cargo/issues/7914 hits stable, we can remove these `features`

Reply:
Strange, this solution works for me just fine:
```toml
[target.'cfg(target_arch = "arm")'.dependencies]
cortex-m = "*"

[target.'cfg(target_arch = "riscv32")'.dependencies]
riscv = "*"
```

Reply:
Yep, it does. Please see the next commit

Reply:
Neither `master` nor this branch work on stable, but both work on beta and nightly just fine.